code/game/dna/genes/goon_powers.dm

```diff
@@ -614,7 +612,7 @@
 		to_chat(user, "<span class='warning'>This can only be used on carbon beings.</span>")
 		return 1
 
-	if (!can_mind_interact(M.mind))
+	if (M_PSY_RESIST in M.mutations)
 		to_chat(user, "<span class='warning'>You can't see into [M.name]'s mind at all!</span>")
```

Comment:
how come tinfoil blocks telepathic messages but doesn't stop the government from READING YOUR MIND and taking your bank PIN

Reply:
tinfoil is intended to block all mind powers

Reply:
wouldnt it neater and overall less retarded if putting on a tinfoil hat just gave you PSY_RESIST?